### ISSUE TITLE
Test script code discrepancy

### ISSUE THREAD

**Issue by pv33** [ivapylibs/puzzle_solver#50]
puzzle/pieces /testing work

**Comment by pv33** [comment]
Package                       Version              Location                       
----------------------------- -------------------- -------------------------------
actionlib                     1.14.0               
alabaster                     0.7.8                
angles                        1.9.13               
apt-xapian-index              0.49                 
atomicwrites                  1.1.5                
attrs                         19.3.0               
autobahn                      17.10.1              
Automat                       0.8.0                
Babel                         2.6.0                
base-local-planner            1.17.3               
bcrypt                        3.1.7                
beautifulsoup4                4.8.2                
blinker                       1.4                  
blosc                         1.7.0                
bokeh                         3.0.3                
bondpy                        1.8.6                
breezy                        3.0.2                
camera                        2.2.0                /home/pvela/python/camera      
camera-calibration            1.17.0               
camera-calibration-parsers    1.12.0               
capabilities                  0.3.1                
catkin                        0.8.10               
catkin-pkg                    1.0.0                
catkin-pkg-modules            1.0.0                
catkin-tools                  0.9.2                
cbor                          1.0.0                
certifi                       2019.11.28           
chardet                       3.0.4                
charset-normalizer            2.1.0                
Click                         7.0                  
cloudpickle                   1.3.0                
colorama                      0.4.3                
colorlog                      6.7.0                
command-not-found             0.3                  
configobj                     5.0.6                
constantly                    15.1.0               
contourpy                     1.0.7                
controller-manager            0.20.0               
controller-manager-msgs       0.20.0               
cryptography                  2.8                  
cupshelpers                   1.0                  
cv-bridge                     1.16.2               
cycler                        0.10.0               
Cython                        0.29.14              
dask                          2.8.1+dfsg           
dataclasses                   0.6                  
dbus-python                   1.2.16               
decorator                     4.4.2                
defusedxml                    0.6.0                
Deprecated                    1.2.13               
descartes                     1.1.0                
detector                      1.0.1                /home/pvela/python/detector    
diagnostic-analysis           1.11.0               
diagnostic-common-diagnostics 1.11.0               
diagnostic-updater            1.11.0               
distro                        1.4.0                
distro-info                   0.23+ubuntu1.1       
docutils                      0.16                 
dulwich                       0.19.15              
dynamic-reconfigure           1.7.3                
empy                          3.3.2                
entrypoints                   0.3                  
et-xmlfile                    1.0.1                
fastimport                    0.9.8                
fsspec                        0.6.1                
ftfy                          6.1.1                
future                        0.18.2               
gazebo-plugins                2.9.2                
gazebo-ros                    2.9.2                
gencpp                        0.7.0                
geneus                        3.0.0                
genlisp                       0.4.18               
genmsg                        0.6.0                
gennodejs                     2.0.2                
genpy                         0.6.15               
gpg                           1.13.1               
h5py                          3.10.0               
html5lib                      1.0.1                
httplib2                      0.14.0               
hyperlink                     19.0.0               
idna                          2.8                  
image-geometry                1.16.2               
imageio                       2.33.1               
imagesize                     1.2.0                
importlib-metadata            1.5.0                
importlib-resources           5.10.2               
improcessor                   1.0.1                /home/pvela/python/improcessor 
incremental                   16.10.1              
interactive-markers           1.12.0               
ivapy                         1.0.1                /home/pvela/python/ivapy       
jdcal                         1.0                  
Jinja2                        2.10.1               
joblib                        1.4.2                
joint-state-publisher         1.15.1               
joint-state-publisher-gui     1.15.1               
jsonschema                    4.17.3               
kdl-parser-py                 1.14.2               
keyring                       18.0.1               
kiwisolver                    1.0.1                
language-selector             0.1                  
laser-geometry                1.6.7                
launchpadlib                  1.10.13              
lazr.restfulclient            0.14.2               
lazr.uri                      1.0.3                
lazy-loader                   0.3                  
Lie                           1.0                  /home/pvela/python/Lie         
locket                        0.2.0                
lxml                          4.5.0                
lz4                           3.0.2+dfsg           
mailchecker                   4.1.17               
mapbox-earcut                 1.0.1                
MarkupSafe                    1.1.0                
matplotlib                    3.1.2                
matplotlib-venn               0.11.5               
message-filters               1.16.0               
more-itertools                4.2.0                
moveit-commander              1.1.14               
moveit-core                   1.1.14               
moveit-python                 0.4.5                
moveit-ros-planning-interface 1.1.14               
moveit-ros-visualization      1.1.14               
mpi4py                        3.0.3                
mpmath                        1.2.1                
msgpack                       1.0.4                
netifaces                     0.10.4               
networkx                      3.1                  
noise                         1.2.2                
nose                          1.3.7                
numexpr                       2.7.1                
numpy                         1.23.5               
oauthlib                      3.1.0                
olefile                       0.46                 
opencv-contrib-python         4.5.5.62             
opencv-python                 4.6.0.66             
openpyxl                      3.0.3                
osrf-pycommon                 2.0.2                
packaging                     23.2                 
pandas                        1.5.3                
paramiko                      2.6.0                
partd                         1.0.0                
pcg-gazebo                    0.7.12               
perceiver                     1.0.1                /home/pvela/python/perceiver   
pexpect                       4.6.0                
phonenumbers                  8.12.51              
Pillow                        10.1.0               
pip                           20.0.2               
pkgutil-resolve-name          1.3.10               
pluggy                        0.13.0               
progress                      1.6                  
psutil                        5.5.1                
py                            1.8.1                
py-ubjson                     0.14.0               
pyasn1                        0.4.2                
pyasn1-modules                0.2.1                
pyassimp                      4.1.4                
pycairo                       1.16.2               
pycollada                     0.6                  
pycrypto                      2.6.1                
pycryptodomex                 3.6.1                
pycups                        1.9.73               
pydot                         1.4.1                
PyGithub                      1.43.7               
Pygments                      2.3.1                
PyGObject                     3.36.0               
PyHamcrest                    1.9.0                
PyJWT                         1.7.1                
pymacaroons                   0.13.0               
PyNaCl                        1.3.0                
PyOpenGL                      3.1.0                
pyOpenSSL                     19.0.0               
pyparsing                     2.4.6                
pypng                         0.0.20               
PyQRCode                      1.2.1                
PyQt5                         5.14.1               
pyrealsense2                  2.50.0.3812          
pyrsistent                    0.19.3               
pyserial                      3.4                  
pytest                        4.6.9                
python-apt                    2.0.1+ubuntu0.20.4.1 
python-benedict               0.25.1               
python-dateutil               2.8.2                
python-debian                 0.1.36+ubuntu1.1     
python-fcl                    0.7.0.4              
python-fsutil                 0.6.1                
python-gitlab                 2.0.1                
python-gnupg                  0.4.5                
python-qt-binding             0.4.4                
python-slugify                6.1.2                
python-snappy                 0.5.3                
python-tsp                    0.3.0                
pytoml                        0.1.21               
PyTrie                        0.2                  
pytz                          2022.7.1             
PyWavelets                    1.4.1                
PyYAML                        6.0                  
pyzmq                         18.1.1               
qt-dotgraph                   0.4.2                
qt-gui                        0.4.2                
qt-gui-cpp                    0.4.2                
qt-gui-py-common              0.4.2                
reportlab                     3.5.34               
requests                      2.28.1               
requests-unixsocket           0.2.0                
resource-retriever            1.12.7               
roipoly                       0.5.3                
roman                         2.0.0                
ros-numpy                     0.0.5                
rosbag                        1.16.0               
rosboost-cfg                  1.15.8               
rosclean                      1.15.8               
roscreate                     1.15.8               
rosdep                        0.25.1               
rosdep-modules                0.25.1               
rosdistro                     0.9.1                
rosdistro-modules             0.9.1                
rosgraph                      1.16.0               
rosinstall                    0.7.8                
rosinstall-generator          0.1.23               
roslaunch                     1.16.0               
roslib                        1.15.8               
roslint                       0.12.0               
roslz4                        1.16.0               
rosmake                       1.15.8               
rosmaster                     1.16.0               
rosmsg                        1.16.0               
rosnode                       1.16.0               
rosparam                      1.16.0               
rospkg                        1.5.1                
rospkg-modules                1.5.1                
rospy                         1.16.0               
rospy-message-converter       0.5.9                
rosservice                    1.16.0               
rostest                       1.16.0               
rostopic                      1.16.0               
rosunit                       1.15.8               
ROSWrapper                    1.0                  /home/pvela/python/ROSWrapper  
roswtf                        1.16.0               
rqt-action                    0.4.9                
rqt-bag                       0.5.1                
rqt-bag-plugins               0.5.1                
rqt-console                   0.4.12               
rqt-dep                       0.4.12               
rqt-graph                     0.4.14               
rqt-gui                       0.5.3                
rqt-gui-py                    0.5.3                
rqt-image-view                0.4.17               
rqt-launch                    0.4.9                
rqt-logger-level              0.4.12               
rqt-moveit                    0.5.11               
rqt-msg                       0.4.10               
rqt-nav-view                  0.5.7                
rqt-plot                      0.4.13               
rqt-pose-view                 0.5.11               
rqt-publisher                 0.4.10               
rqt-py-common                 0.5.3                
rqt-py-console                0.4.10               
rqt-reconfigure               0.5.5                
rqt-robot-dashboard           0.5.8                
rqt-robot-monitor             0.5.15               
rqt-robot-steering            0.5.12               
rqt-runtime-monitor           0.5.10               
rqt-rviz                      0.7.0                
rqt-service-caller            0.4.10               
rqt-shell                     0.4.11               
rqt-srv                       0.4.9                
rqt-tf-tree                   0.6.4                
rqt-top                       0.4.10               
rqt-topic                     0.4.13               
rqt-web                       0.4.10               
Rtree                         1.0.1                
rviz                          1.14.25              
scikit-image                  0.21.0               
scikit-learn                  1.3.2                
scipy                         1.8.1                
screen-resolution-extra       0.0.0                
SecretStorage                 2.3.1                
sensor-msgs                   1.13.1               
service-identity              18.1.0               
setuptools                    45.2.0               
Shapely                       1.7.0                
similaritymeasures            1.1.0                
simplejson                    3.16.0               
sip                           4.19.21              
six                           1.14.0               
sklearn                       0.0                  
sklearn-pandas                1.8.0                
smach                         2.5.2                
smach-ros                     2.5.2                
smclib                        1.8.6                
soupsieve                     1.9.5                
Sphinx                        1.8.5                
srdfdom                       0.6.4                
ssh-import-id                 5.10                 
Surveillance                  1.0                  /home/pvela/python/surveillance
svg.path                      6.2                  
sympy                         1.11.1               
systemd-python                234                  
tables                        3.6.1                
tabulate                      0.8.10               
text-unidecode                1.3                  
tf                            1.13.2               
tf-conversions                1.13.2               
tf2-geometry-msgs             0.7.7                
tf2-kdl                       0.7.7                
tf2-py                        0.7.7                
tf2-ros                       0.7.7                
threadpoolctl                 3.5.0                
tifffile                      2023.7.10            
toml                          0.10.2               
toolz                         0.9.0                
topic-tools                   1.16.0               
tornado                       6.2                  
tqdm                          4.64.0               
trackpointer                  1.0.1                /home/pvela/python/trackpointer
transitions                   0.8.11               
triangle                      20220202             
trimesh                       3.19.4               
tsplib95                      0.7.1                
Twisted                       18.9.0               
txaio                         2.10.0               
u-msgpack-python              2.1                  
ubuntu-advantage-tools        8001                 
ubuntu-drivers-common         0.0.0                
ufw                           0.36                 
unattended-upgrades           0.1                  
urdfdom-py                    0.4.6                
urllib3                       1.25.8               
usb-creator                   0.3.7                
vcstools                      0.1.42               
wadllib                       1.3.3                
wcwidth                       0.2.5                
webencodings                  0.5.1                
wheel                         0.34.2               
wrapt                         1.14.1               
wsaccel                       0.6.2                
wstool                        0.1.18               
xacro                         1.14.18              
xkit                          0.0.0                
xlrd                          1.1.0                
xlwt                          1.3.0                
xmltodict                     0.13.0               
xxhash                        3.2.0                
xyzservices                   2022.9.0             
yacs                          0.1.8                
yasha                         5.0                  
zipp                          3.13.0               
zope.interface                4.7.1

**Comment by pv33** [comment]
@dvsh-n 
I got the `puzzle/testing/clusters` and `testing/pieces/testing` directories to have mostly functional code.  What works is checked off in the list at the top of the git issue.  Everything should run. If it doesn't, then maybe there are library/package differences.  I posted by package listing to this issue in a comment.  Worth comparing.

It would be a good exercise to get the SIFT feature working.  I never upgraded the class code for SIFT.  Yunzhi liked to do `shapeExtractFeat` and `colorExtractFeat` or maybe FeatExtract, who knows.  All I know is that it is now called `extractFeature`.  you can use the histgoram of other coded and functional methods to see how to code the SIFT version.  There's enough working that you should be able to get the revisions made pretty quickly.
